### PR TITLE
LPS-144903 Refactor DatePicker

### PR DESCRIPTION
**Pull request by miltonmc**
### Ticket
[LPS-144903](https://issues.liferay.com/browse/LPS-144903) Refactors DatePicker

### Related Issues
- [LPS-140281](https://issues.liferay.com/browse/LPS-140281) Predefined value on date field not working with Japanese language
- [LPS-141491](https://issues.liferay.com/browse/LPS-141491) The date field predefined value has an inconsistent behaviour between sidebar and form builder when using translations
- [LPS-145009](https://issues.liferay.com/browse/LPS-145009) Hungarian translation Date Field isn't working with date validation
- [LPS-145053](https://issues.liferay.com/browse/LPS-145053) Sometimes the Date is shown blank when using the datepicker

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @miltonmc, can you please check if these failures are related to your changes? Thanks

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js [hunk 1]
```diff
@@ -26,75 +26,24 @@ const LETTER_DIGIT_REGEX = /[A-Z0-9]/gi;
...
-
-	if (dateFormat.indexOf('-') !== -1) {
-		dateDelimiter = '-';
-	}
-
-	return dateDelimiter;
-};
-
-const getLocaleDateFormat = (locale, format = 'L') => {
+const getLocaleDateFormat = (locale) => {
```
Comment (rebsilva) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:29]: Hi @miltonmc! Awesome changes you've made, but I have a question about this part: why did you remove the "format" parameter here? Because for exemple, on the task of the story I'm working on, I do something like this:

```
const date = momentLocale.localeData().longDateFormat('L');
const time = momentLocale.localeData().longDateFormat('LT');
```
So if the function had the "format" parameter I could pass the 'L' or 'LT' in this case, doing something like this:

```
const date = getLocaleDateFormat('L');
const time = getLocaleDateFormat('LT');
```

Wouldn't this be better or is not really necessary since it's a simple case?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es.js [hunk 2]
```diff
@@ -390,7 +388,6 @@ function FieldBase({
 
 			{!hideEditedFlag && (
 				<input
```
Comment (rebsilva) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es.js:390]: Could you explain why did you remove the key parameter here?

**Review by rebsilva — COMMENTED** [review]
Great job @miltonmc! I just made some questions but everything looks fine to me!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es.js (hunk 2 above)
Comment (miltonmc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es.js:390]: We should only use the `key` attribute for element into an iteration (like an array)

The React documentation better explain that:

> A “key” is a special string attribute you need to include when creating lists of elements.

> Keys help React identify which items have changed, are added, or are removed. Keys should be given to the elements inside the array to give the elements a stable identity

https://reactjs.org/docs/lists-and-keys.html

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js (hunk 1 above)
Comment (miltonmc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:29]: I've removed the parameter, because it was not been called anywhere.

You will probably need to add a new parameter to specify if the format will contain the time or not, but not to specify the format (as the one I've removed). And you got it right, you will need to do something like you've shown here, to get the time format.

**Comment by miltonmc** [comment]
ci:test:sf

**Comment by miltonmc** [comment]
ci:test:relevant

**Comment by miltonmc** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @miltonmc, can you please check if these failures are related to your changes? Thanks

**Comment by miltonmc** [comment]
ci:reevaluate:1297401_1325

**Comment by liferay-forms** [comment]
Hey @miltonmc, can you please check if these failures are related to your changes? Thanks

**Comment by miltonmc** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by miltonmc** [comment]
ci:forward
